File: 02-calculations/01-square-area/solution.js
import readline from 'readline/promises';
import {stdin as input, stdout as output} from 'process';

const rl = readline.createInterface({
    input, output
});


async function getSide(){
    let side = (await rl.question("Enter the value of the side of the square: ")).trim();
    side = parseFloat(side);

    if (!isNaN(side) && side > 0){
        return side;
    }   
    
    return 0.0;
}


async function main(){
    const side = await getSide();
    if (side == 0.0){
        console.log('Invalid side value.');
    } else {
        const area = side * side;
        console.log(`Square area is ${area}`);
    }
    rl.close();
}

main()

File: 02-calculations/01-square-area/solution.c
#include <stdio.h>

float get_side(void);

int main(void)
{
    float side = get_side();    
    if (side == 0.0)
    {
        printf("Invalid side value.\n");
        return 0;
    }

    float area = side * side;
    printf("Square area is %f\n", area);

    return 0;
}

float get_side(void)
{
    printf("Enter the value of the side of the square: ");

    float side;

    if (scanf("%f", &side) != 1)
    {
        return 0.0;
    }

    if (side <= 0)
    {
        return 0.0;
    }

    return side;
}